common/src/main/java/org/apache/drill/common/scanner/RunTimeScan.java

```diff
@@ -75,4 +77,20 @@ public static ScanResult fromPrescan(DrillConfig config) {
     }
   }
 
+  /**
+   * Scanning packages at runtime only for marked paths and list of pre-scanned annotations
+   * @param config to retrieve the packages to scan
+   * @param markedPath list of paths where to scan
+   * @return the scan result
+   */
+  public static ScanResult dynamicPackageScan(DrillConfig config, Set<URL> markedPath) {
```

Comment:
Can you explain the purposes of this a bit more? What is a "marked path"? What is a "prescanned annotation?" At a higher level, what service does this provide? Allows me to locate (what?) given (what?)...